src/engine/QueryExecutionTree.cpp

```diff
@@ -471,44 +403,57 @@ nlohmann::json QueryExecutionTree::writeQLeverJsonTable(
 
 // _____________________________________________________________________________
 template <QueryExecutionTree::ExportSubFormat format>
-ad_utility::stream_generator::stream_generator QueryExecutionTree::writeTable(
-    size_t from, size_t upperBound,
-    const vector<std::optional<pair<size_t, ResultTable::ResultType>>>
-        validIndices,
-    std::shared_ptr<const ResultTable> resultTable) const {
+ad_utility::stream_generator::stream_generator
+QueryExecutionTree::generateResults(
+    const SelectedVarsOrAsterisk& selectedVarsOrAsterisk, size_t limit,
+    size_t offset) const {
   static_assert(format == ExportSubFormat::BINARY ||
                 format == ExportSubFormat::CSV ||
                 format == ExportSubFormat::TSV);
-  if (!resultTable) {
-    resultTable = getResult();
-  }
+
+  // They may trigger computation (but does not have to).
```

Comment:
// This call triggers the possibly expensive computation of the query result unless the result is already cached.

(I known this wasn't your comment).